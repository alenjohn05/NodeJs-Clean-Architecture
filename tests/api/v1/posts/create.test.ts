import { client } from '../..';

describe('POST /api/v1/posts', () => {
  describe('given an invalid request', () => {
    test('responds with a 400 status code', async () => {
      const response = await client.post('/api/v1/posts').send({
        published: true,
        title: '', // Cannot be empty
      });

      expect(response.status).toEqual(400);
    });
  });

  describe('given a valid request', () => {
    test('creates post and responds with a 201 status code', async () => {
      const createResponse = await client.post('/api/v1/posts').send({
        published: true,
        title: 'Example post',
      });

      const getResponse = await client.get(`/api/v1/posts/${createResponse.body.id}`);

      expect(createResponse.status).toEqual(201);
      expect(createResponse.body).toMatchSnapshot({ id: expect.any(Number) });
      expect(getResponse.status).toEqual(200);
    });
  });
});
