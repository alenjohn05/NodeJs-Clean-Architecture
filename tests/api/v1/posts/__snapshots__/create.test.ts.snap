// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`POST /api/v1/posts given a valid request creates post and responds with a 201 status code 1`] = `
{
  "id": Any<Number>,
}
`;
